src/log_file.rs

```diff
@@ -86,3 +99,16 @@ fn clean_filename(s: &str) -> String {
         })
         .collect::<String>()
 }
+
+#[cfg(test)]
+mod test {
+    use crate::log_file::{ECRYPTFS_FILE_NAME_LENGTH_LIMIT, LogFile};
+
+    #[test]
+    fn should_reduce_file_name_length_below_ecryptfs_limits() {
+        let basename = "X".repeat(ECRYPTFS_FILE_NAME_LENGTH_LIMIT+1);
+        let name = LogFile::log_file_name(&basename, 0);
```

Comment:
I'd like to see this instead call `LogFile::create` in a tempdir, and it should do it 2x with the same long name, to check that they do actually get unique names.